File: update_ShotList.m
function update_ShotList

global SA

hsl = findobj('Tag', 'uitable_shotlist');


[rowshow, roworder] = ismember(cell2mat(SA.ShotIDsVisible), cell2mat(SA.Table.ShotID)+cell2mat(SA.Table.Mirrored)/10);
[colshow, colorder] = ismember(SA.ColumnsVisible, SA.Table.Properties.VariableNames);

data = table2array(SA.Table(roworder(rowshow),colorder(colshow)));


%% Update ShotList
set(hsl,'ColumnWidth','auto', 'ColumnName', SA.ColumnsVisible, ...
    'Data', data , ...
    'ColumnEditable',true);

File: update_shot_list.py
def update_shot_list(SA):
    """
    Update the shot list in the GUI based on the current state of SA.
    """
    # Extract visible rows and columns
    rowshow = [i for i, shot_id in enumerate(SA['ShotIDsVisible']) if shot_id in [sid + mir / 10 for sid, mir in zip(SA['Table']['ShotID'], SA['Table']['Mirrored'])]]
    colshow = [col for col in SA['ColumnsVisible'] if col in SA['Table'].columns]

    # Filter data for visible rows and columns
    data = SA['Table'].iloc[rowshow][colshow]

    # Update the GUI table (placeholder for actual GUI update logic)
    print("Updated shot list with the following data:")
    print(data)

    # If integrated with a GUI framework, replace the print statement with actual table update logic.